perses/tests/test_repex.py

```diff
@@ -188,5 +189,229 @@ def test_RESTCapableHybridTopologyFactory_repex_charge_mutation():
+            reporter = MultiStateReporter(
+                reporter_file,
+                analysis_particle_indices=htf.hybrid_topology.select(selection),
+                checkpoint_interval=100)
+
+            # Build the hybrid repex sampler
+            sampler = HybridRepexSampler(
+                mcmc_moves=mcmc.LangevinDynamicsMove(
+                    timestep=4.0 * unit.femtoseconds,
+                    collision_rate=1.0 / unit.picosecond,
+                    n_steps=250,
+                    reassign_velocities=False,
+                    n_restart_attempts=20,
+                    constraint_tolerance=1e-06),
+                replica_mixing_scheme='swap-all',
+                hybrid_factory=htf,
+                online_analysis_interval=None)
+
+            sampler.setup(
+                n_states=n_states,
+                temperature=300 * unit.kelvin,
+                storage_file=reporter,
+                minimisation_steps=0,
+                endstates=True)
+
+            sampler.energy_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+            sampler.sampler_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+
+            # Run repex
+            sampler.extend(n_iterations)
+
+            # Retrieve free energy
+            reporter.close()
+            reporter = MultiStateReporter(reporter_file)
+            analyzer = MultiStateSamplerAnalyzer(reporter, max_n_iterations=n_iterations)
+            f_ij, df_ij = analyzer.get_free_energy()
+            data[f"{ligand_A_index}-{ligand_B_index}"] = {'free_energy': f_ij[0, -1], 'error': df_ij[0, -1]}
+
+        DDG = abs(data['0-1']['free_energy'] - data['1-0']['free_energy'] * -1)
+        dDDG = np.sqrt(data['0-1']['error'] ** 2 + data['1-0']['error'] ** 2)
+        assert DDG < 6 * dDDG, f"DDG ({DDG}) is greater than 6 * dDDG ({6 * dDDG})"
+
+
+@pytest.mark.skip(reason="Need better system for test")
+@pytest.mark.gpu_needed
+def test_RESTCapableHybridTopologyFactory_repex_charge_transformation():
+    """
+    Run repex on a host-guest charge-changing transformation (CB7:A1->A2 and CB7:A2->A1) with the
+    RESTCapableHybridTopologyFactory and make sure that the free energies are equal and opposite.
+
+    """
+
+    from pkg_resources import resource_filename
+    import numpy as np
+
+    from openmm import unit
+
+    from perses.dispersed.utils import configure_platform
+    from perses.app.relative_setup import RelativeFEPSetup
+    from perses.samplers.multistate import HybridRepexSampler
+    from perses.annihilation.relative import RESTCapableHybridTopologyFactory
+
+    from openmmtools.multistate import MultiStateReporter, MultiStateSamplerAnalyzer
+    from openmmtools import cache, utils, mcmc
+    platform = configure_platform(utils.get_fastest_platform().getName())
 
+    data = {}
+    n_iterations = 2000
+    n_states = 12
```

Comment:
The charge-changing mutation test requires 36 states to pass, so I wonder if we need to increase to 36 states here as well to get the test to pass.

Reply:
Did you make sure to bootstrap the uncertainties? That might help make the test past. Otherwise, I agree with you, might be better to use tyk2 ligands.

Reply:
You are right, this was without bootstrapping. I'll check with bootstrapping.